### PR TITLE
Add support for test framework to run with local Kind cluster

### PR DESCRIPTION
**Pull request by antontroshin**
# Description

Allows for the pert tests (and e2e?) to run seamlessly with the local Kind cluster without modifying test "app URL" etc.
## Issue reference

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by cicoyle — DISMISSED** [review]
I think this is a nice improvement, @antontroshin mind updating the README to mention the `kind-kind` cluster name or how someone might override that name with a diff cluster name in kind?

File: tests/docs/running-perf-tests.md [hunk 1]
```diff
@@ -10,7 +10,8 @@ Performance tests are designed to let you evaluate the latency, resource usage a
 ### Prerequisites
 
 * Kubernetes cluster (Minikube and Kind are valid options too).
-  - To setup a new Kind cluster and local registry, run `make setup-kind`.
+  - To setup a new Kind cluster and local registry, run `make setup-kind`. 
+  - **NOTE**: `make setup-kind` command will create a new kind cluster with default name `kind-kind`. This name is used to automatically determine if the tests are running locally in a kind cluster or not. If you want to use a different name, you can set the `DAPR_TEST_KIND_CLUSTER_NAME` environment variable. For example, if you want to use `kind-mycluster` as the name of the kind cluster, you can run `export DAPR_TEST_KIND_CLUSTER_NAME=kind-mycluster` before running `make setup-kind` and later to run the tests.
```
Comment (elena-kolevska) [inline tests/docs/running-perf-tests.md:14]:
```suggestion
  - **NOTE**: `make setup-kind` command will create a new kind cluster with the default name `kind`. This name is used to automatically determine if the tests are running locally in a kind cluster or not. If you want to use a different name, you can set the `DAPR_TEST_KIND_CLUSTER_NAME` environment variable. For example, if you want to use `kind-mycluster` as the name of the kind cluster, you can run `export DAPR_TEST_KIND_CLUSTER_NAME=kind-mycluster` before running `make setup-kind` and later to run the tests.
```

**Review by elena-kolevska — CHANGES_REQUESTED** [review]
The `stable` repo is not maintained anymore, I was getting this error on metrics server install:
```
# Setup metrics-server
helm install ms stable/metrics-server -n kube-system --set=args={--kubelet-insecure-tls}
Error: INSTALLATION FAILED: repo stable not found
```

Updating to `kubernetes-sigs.github.io/metrics-server/` fixed it:
```
helm repo add metrics-server https://kubernetes-sigs.github.io/metrics-server/
helm upgrade --install ms metrics-server/metrics-server -n kube-system --set args={--kubelet-insecure-tls}
```

Also, I had to add `export TARGET_ARCH=arm64` before `make build-linux`, otherwise I was getting the error:
`ERROR: unable to prepare context: path "./dist/linux_amd64/release" not found
make: *** [docker-build] Error 1`

With these changes and the ones I shared below - I was able to get the perf tests running locally! Thanks for making that possible @antontroshin

**Comment by antontroshin** [comment]
@elena-kolevska The metrics server command fix was addressed in another PR https://github.com/dapr/dapr/pull/7992
Thanks for checking this out as well.

**Review by JoshVanL — APPROVED** [review]

**Review by elena-kolevska — DISMISSED** [review]

**Review by artursouza — APPROVED** [review]

**Pull request merged**